### PR TITLE
[WIP] ZIOS-10527: User suggestions popup

### PR DESCRIPTION
**Pull request by nicorsm**
## What's new in this PR?

This is still work in progress, added support for the user suggestions popup.

*Blocking tasks*
- Collection view cells are not visible (cell size and `cellForItemAt:` methods don't get called)

*To-do*
- Implement search algorithm (see docs)
- Attach to the input bar (call `show()` when needed, `dismiss()` when selecting a user)
- Adapt collection view height based on device size
- Check if data is correct
- Write more tests

### CODE REVIEW

**Comment by vytis** [comment]
We have a `feature/mentions` branch where this feature lives for now - better to merge it there

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 1]
```diff
@@ -116,7 +116,7 @@
          </CommandLineArgument>
          <CommandLineArgument
             argument = "-ZMBackendEnvironmentType staging"
-            isEnabled = "NO">
+            isEnabled = "YES">
```
Comment (mikeger) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme]: Better not to!

File: Wire-iOS Tests/MentionsSearchResultsViewControllerTests.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
+//
+//  MentionsSearchResultsViewControllerTests.swift
+//  Wire-iOS-Tests
+//
+//  Created by Nicola Giancecchi on 12.09.18.
+//  Copyright © 2018 Zeta Project Germany GmbH. All rights reserved.
+//
```
Comment (mikeger) [inline Wire-iOS Tests/MentionsSearchResultsViewControllerTests.swift]: Missing header

File: Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,133 @@
+//
+//  MentionsSearchResultsViewController.swift
+//  Wire-iOS
+//
+//  Created by Nicola Giancecchi on 12.09.18.
+//  Copyright © 2018 Zeta Project Germany GmbH. All rights reserved.
+//
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift]: Missing header.

File: Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    @objc func searchForUsers(with name: String) {
+        pendingSearchTask?.cancel()
+        
+        let request = SearchRequest(query: query,
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift]: I think we should not include the users that are not members of this conversation, so we can just filter the conversation.participants array.

File: Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    
+    func cancelPreviousSearch() {
+        pendingSearchTask?.cancel()
+        pendingSearchTask = nil
+    }
+    
+    func show() {
+        self.view.alpha = 0.0
+        self.view.isHidden = false
+        UIView.animate(withDuration: 0.25) {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift]: It was decided to use no animation on the appearance, however this short one might be OK.

File: Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift (hunk 4 above)
Comment (cardassian-tailor) [reply Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift]: > 
> 
> I think we should not include the users that are not members of this conversation, so we can just filter the conversation.participants array.

I was thinking about this. Has anyone conducted a test on what occurs if you mention a user whom isnt within the current conversation? How about a user your not connected with?

File: Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        
+        constrain(self.view, collectionView) { (selfView, collectionView) in
+            collectionView.bottom == selfView.bottom
+            collectionView.leading == selfView.leading
+            collectionView.trailing == selfView.trailing
+            tableViewHeight = collectionView.height == 0
+        }
+    }
+
+    override func didReceiveMemoryWarning() {
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Conversation/Mentions/MentionsSearchResultsViewController.swift]: Can be removed?

**Comment by nicorsm** [comment]
@billypchan yes, I need to take a look at that.

File: Wire-iOS Tests/MentionsSearchResultsViewControllerTests.swift [hunk 7]
```diff
@@ -15,21 +25,39 @@ class MentionsSearchResultsViewControllerTests: CoreDataSnapshotTestCase {
     
     override func setUp() {
         super.setUp()
-        sut = MentionsSearchResultsViewController(nibName: nil, bundle: nil)
+        self.recordMode = true
```
Comment (daehn) [inline Wire-iOS Tests/MentionsSearchResultsViewControllerTests.swift]: Should be reset?

**Pull request merged**